transcom/mymove
# [ MB-15891 ] Validate postal code on change in residential address form

## [Jira ticket](https://dp3.atlassian.net/browse/MB-15891)

## Summary

This PR. adds validation to the zip code field on the Current pickup address form for each keypress. 

I did not choose to set the validateOnChange to true for the entire address form, because this allows the ability to spam requests to the `rate_engine_postal_codes` endpoint where we do backend validation on the postal codes. It also would have required some of the same changes to set the field to "touched".

Instead, I added an optional onChange function to the postal code field in the AddressFields component. In it, I use the setFieldTouched Formik function to validate the field and set it to "touched" in order to show any errors. "Touched" is a state used by Formik to indicate that a field has been visited. Typically, Formik only modifies this state when the onBlur event is fired, after shifting focus away from the field.

Please give extra scrutiny to the name `formikFunctionsToValidatePostalCodeOnChange`. It is truly a monster of a name, but it is accurate and descriptive. If you can think of something better I would appreciate it. 

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Access the MilMove app as a customer
2. Create a move.
3. When you get to the current pickup address and the backup address you should notice that the zip field is validating for each key press. 

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
